scripts/doc_gen.py

```diff
@@ -0,0 +1,375 @@
+"""
+This module provides functionality to write a man page from some given
+information about a CLI application and is modified from click-man.
+
+https://github.com/click-contrib/click-man/tree/3a0684c9d70f94a2abef08fb3f88449dab6aea98
+
+The original click-man source is licensed under the MIT License. Changes are
+licensed under the Kart license (GPL).
+"""
+
+import platform
+from pathlib import Path
+import sys
+import time
+import typing as t
+
+import click
+import rst2txt
+from inspect import getmembers
+from importlib import import_module
+from docutils.core import publish_string
+from docutils.writers import manpage
+
+try:
+    from importlib.metadata import distribution
+except ImportError:
+    try:
+        from importlib_metadata import distribution
+    except ImportError:
+        from pkg_resources import load_entry_point
```

Comment:
This should never happen since we're depending on `importlib-metadata` in requirements.txt. Maybe move that into requirements.in to make it explicit? With a comment noting it's only needed for python 3.7